### PR TITLE
Add installation doc

### LINKED ISSUE #104: Write installation docs
Add to Readme.rst that libraries certifi and urllib3 are prerequisites, and add instructions on how to install these @vdumoulin.

### PR DESCRIPTION
**Pull request by lamblin**
Should fix #104, at least in part.

These instructions will use the latest version of picklable-itertools on PyPI, rather than the latest GitHub one. This seems to work fine at the moment, but I can change that if needed.

### CODE REVIEW

**Comment by lamblin** [comment]
@dwf Could you have a look?

File: docs/overview.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ sight, but can be very powerful.
    RequestIterator -> DataIterator [style=dashed, label=" Argument to"];
    DataIterator -> DataStream [label=" Gets data from"];
    DataStream -> DataStream [style=dashed, label=" Gets data from (wrapper)"];
-   { rank=same; RequestIterator, DataIterator }
+   { rank=same; RequestIterator DataIterator }
```
Comment (dwf) [inline docs/overview.rst]: No idea what this is but if it's necessary, ok

File: docs/setup.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
+Installation
+============
+
+The easiest way to install Fuel is using the Python package manager pip. Fuel
```
Comment (dwf) [inline docs/setup.rst]: Might want to use 

   ``pip``

File: docs/setup.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+------------
+Fuel's requirements are
+
+* PyYAML_, to parse the configuration file
+* six_, to support both Python 2 and 3 with a single codebase
+* h5py_ and PyTables_ for the HDF5 storage back-end
+* pillow_, providing PIL for image preprocessing
+* Cython_, for fast extensions
+* pyzmq_, to send data across processes
+* picklable_itertools_, for picklable iterators
```
Comment (dwf) [inline docs/setup.rst]: I'd just say 'for supporting iterator serialization'

File: docs/setup.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+Requirements
+------------
+Fuel's requirements are
+
+* PyYAML_, to parse the configuration file
+* six_, to support both Python 2 and 3 with a single codebase
+* h5py_ and PyTables_ for the HDF5 storage back-end
+* pillow_, providing PIL for image preprocessing
+* Cython_, for fast extensions
+* pyzmq_, to send data across processes
```
Comment (dwf) [inline docs/setup.rst]: to efficiently* send data across processes

File: docs/setup.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+* picklable_itertools_, for picklable iterators
+* SciPy_, to read from MatLab's .mat format
```
Comment (dwf) [inline docs/setup.rst]: It's usually styled MATLAB

File: docs/setup.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+* requests_, to download canonical datasets
+
+.. _Anaconda: https://store.continuum.io/cshop/anaconda/
+.. _nose2: https://nose2.readthedocs.org/
```
Comment (dwf) [inline docs/setup.rst]: You don't mention this anywhere. Probably best to say it's optional "to run the tests".

File: docs/setup.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+   $ pip install -e file:.#egg=fuel[test,docs]
+
+Fuel contains Cython extensions, which need to be recompiled if you
+update the Cython `.pyx` files. Each time these files are modified, you
+should run:
+
+.. code-block:: bash
+
+   $ python setup.py build_ext --inplace
```
Comment (dwf) [inline docs/setup.rst]: Have you verified this works as expected with `pip install -e`? (I'm pretty sure it must.) If so, then great.

File: docs/setup.rst [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+.. _fork Fuel on GitHub: https://github.com/mila-udem/fuel/fork
+
+Documentation
+~~~~~~~~~~~~~
+
+If you want to build a local copy of the documentation, you can follow
+the instructions at `Blocks' documentation development guidelines`_,
+replacing "Blocks" by "Fuel".
```
Comment (dwf) [inline docs/setup.rst]: We should probably just change the Blocks docs to include "Blocks and Fuel" everywhere that mentions "Blocks".

File: docs/overview.rst (hunk 1 above)
Comment (lamblin) [inline docs/overview.rst]: No idea either, but it fixes a parse error...

**Comment by dwf** [comment]
No further comments, and nothing too critical. Fixing them would  be nice. Otherwise, merge at will.

**Comment by lamblin** [comment]
OK, I'll fix that today or tomorrow.

File: docs/setup.rst (hunk 7 above)
Comment (lamblin) [inline docs/setup.rst]: Yes, I checked and that works.

File: docs/setup.rst (hunk 8 above)
Comment (lamblin) [inline docs/setup.rst]: https://github.com/mila-udem/blocks/pull/774

**Comment by lamblin** [comment]
PR updated with the suggested fixes.

**Pull request merged**